Add a bindSecure method

natebosch opened the pull request:
Closes dart-lang/http#1426

natebosch committed:
Add a bindSecure method

Closes #17

natebosch commented on test/http_multi_server_test.dart:
```diff
@@ -237,6 +237,65 @@ void main() {
+      });
+
+      if (await supportsIPv4) {
+        expect(http.read("https://127.0.0.1:${server.port}/"),
```
These fail:

```
HandshakeException: Connection terminated during handshake
```

I think we need something other than `SecurityContext.defaultContext`, or maybe we need to do something different for the `http.read` call... In any case I'm not sure the right way to test this.

natebosch committed:
Fix for new lints

kevmoo commented:
@natebosch – any more thoughts here?

natebosch commented:
> any more thoughts here?

Mainly that I wish I knew how to test it...

grouma commented on lib/http_multi_server.dart:
```diff
@@ -156,6 +156,30 @@ class HttpMultiServer extends StreamView<HttpRequest> implements HttpServer {
+          backlog: backlog, v6Only: v6Only, shared: shared);
+    }
+    if (address == 'any') {
+      return HttpServer.bindSecure(InternetAddress.anyIPv6, port, context,
```
We likely won't want to default to `anyIPv6` here: see https://github.com/dart-lang/http_multi_server/pull/36

natebosch committed:
Merge branch 'master' into bind-secure

natebosch committed:
Check for ipv6 support

natebosch committed:
Fix SSL tests

Copy the SSL certificate and key bytes used in the package:shelf tests.

I am unable to figure out the right `openssl req` command to generate a
cert and key that works without hostname errors.

natebosch committed:
3.0.1 is published, split changelog
